plugins/modules/parcel.py

```diff
@@ -0,0 +1,224 @@
+        super(ClouderaParcel, self).__init__(module)
+
+        self.cluster_name = self.get_param("cluster_name")
+        self.product = self.get_param("product")
+        self.parcel_version = self.get_param("parcel_version")
+        self.state = self.get_param("state")
+
+        self.process()
+
+
+
+    def download_parcel(self, parcel_api_instance, cluster_name, product, parcel_version, polling_interval):
+        parcel_api_instance.start_download_command(cluster_name=cluster_name, product=product, version=parcel_version)
+        while True:
+            parcel_status = parcel_api_instance.read_parcel(cluster_name=cluster_name, product=product, version=parcel_version)
+            if parcel_status.stage == 'DOWNLOADING':
+                time.sleep(polling_interval)
+            elif parcel_status.stage == "DOWNLOADED":
+                break
+            
+    def distribute_parcel(self, parcel_api_instance, cluster_name, product, parcel_version, polling_interval):
+        parcel_api_instance.start_distribution_command(cluster_name=cluster_name, product=product, version=parcel_version)
+        while True:
+            parcel_status = parcel_api_instance.read_parcel(cluster_name=cluster_name, product=product, version=parcel_version)
+            if parcel_status.stage == 'DISTRIBUTING':
+                time.sleep(polling_interval)
+            elif parcel_status.stage == "DISTRIBUTED":
+                break
+
+    def activate_parcel(self, parcel_api_instance, cluster_name, product, parcel_version, polling_interval):
+        parcel_api_instance.activate_command(cluster_name=cluster_name, product=product, version=parcel_version)
+        while True:
+            parcel_status = parcel_api_instance.read_parcel(cluster_name=cluster_name, product=product, version=parcel_version)
+            if parcel_status.stage == 'ACTIVATING':
+                time.sleep(polling_interval)
+            elif parcel_status.stage == "ACTIVATED":
+                break
+
+
+    @ClouderaManagerModule.handle_process
+    def process(self):
+        parcel_api_instance = ParcelResourceApi(self.api_client)
+        cluster_api_instance = ClustersResourceApi(self.api_client)
+
+        polling_interval = 10
+        self.parcel_output  = {}
+        self.changed = False
+        parcel_actions = []
+
+        try:
+            cluster_api_instance.read_cluster(cluster_name=self.cluster_name).to_dict()
+        except ApiException as ex:
+            if ex.status == 404:
+                self.module.fail_json(msg=f" Cluster {self.cluster_name} {ex.reason}")
+
+        try: 
+            existing_state = parcel_api_instance.read_parcel(cluster_name=self.cluster_name, product=self.product, version=self.parcel_version).stage
+        except ApiException as ex:
+                if ex.status == 404:
+                    self.module.fail_json(msg=f" Parcel {self.parcel_version} {ex.reason}")
+
+
+        if self.state == "download":
```

Comment:
States should be declarative - `downloaded`, `activated`, etc.

Also need to add states for `removed`, `undistributed`, and `deactivated`.